### DISCUSSION TITLE
NetUnitPrice - Nachkommastellen

### DISCUSSION THREAD

**Discussion by TietjenMichael** [stephanstapel/ZUGFeRD-csharp#526] [General]
Ich benötige bei NetUnitPrice 3 Nachkommastellen. 
Bei AddTradeLineItem wird der Betrag korrekt mit 3 Nachkommastellen ( 0,044) übergeben. 

In der XML jedoch erschient dieser wie folgt:
        <ram:NetPriceProductTradePrice>
          <ram:ChargeAmount>**0.04**</ram:ChargeAmount>
        </ram:NetPriceProductTradePrice>

Was übersehe ich hier, damit der Betrag mit 3 Nachkomastellen in die XML Datei geschrieben wird ?

Vielen Dank und ein gesundes neues Jahr

**Comment by stephanstapel** [comment]
Hallo @TietjenMichael ,

hier ist eine Diskussion und ein Screenshot aus der Dokumentation:
https://github.com/stephanstapel/ZUGFeRD-csharp/issues/398#issuecomment-2532195957

**Comment by TietjenMichael** [comment]
hallo @stephanstapel 

vielen Dank für die schnelle Info.
Jedoch löst dies leider mein Problem nicht. ich formatiere NetUnitPrice korrekt mit 3 Nachkommastellen, die XML jedoch wird nur mit 2 Nachkommastellen geschrieben.
Was übersehe ich  hier ?

**Reply by tobitege** [reply]
> hallo @stephanstapel
> 
> vielen Dank für die schnelle Info. Jedoch löst dies leider mein Problem nicht. ich formatiere NetUnitPrice korrekt mit 3 Nachkommastellen, die XML jedoch wird nur mit 2 Nachkommastellen geschrieben. Was übersehe ich hier ?

Der Standard erlaubt für `ChargeAmount` halt nur 2 Nachkommastellen. 

Ich habe mal o1 gebeten, eine alternative Angabe der Werte vorzuschlagen:

1. Passe den Wert von „BasisQuantity“ an, um die erweiterte Genauigkeit zu beibehalten.  
   Wenn du den Preis mit drei Nachkommastellen angeben möchtest (z. B. 0,444), das ZUGFeRD-Format jedoch nur zwei Nachkommastellen in „ChargeAmount“ erlaubt, kannst du den „ChargeAmount“ vergrößern und den Wert der „BasisQuantity“ entsprechend verkleinern, sodass sich der Gesamtpreis nicht ändert.  
   - Erhöhe „ChargeAmount“ (z. B. durch Multiplikation mit 1000), sodass keine unerlaubten Nachkommastellen mehr vorliegen.  
   - Verringere „BasisQuantity“ im gleichen Verhältnis (z. B. durch Division durch 1000).  
   So bleibt der Zeilengesamtbetrag unverändert, auch wenn die Werte nun ungewohnt erscheinen.

2. Beispiel „Davor“ und „Danach“.  
   - Angenommen, ursprünglich sollte der Preis 0,444 und die Menge 1 betragen (Zeilengesamtbetrag = 0,444). Diese Angabe verstößt jedoch gegen die ZUGFeRD-Vorgabe von maximal zwei Nachkommastellen in „ChargeAmount“.  
   - Eine Lösung besteht darin, den Preis auf 444 (d. h. Multiplizieren mit 1000) zu setzen und die Menge auf 0,001 zu reduzieren (d. h. Division durch 1000). Mathematisch ergibt sich derselbe Zeilengesamtbetrag:  
     • NetPriceProductTradePrice = 444 × 0,001 = 0,444

**Reply by TietjenMichael** [reply]
hallo @tobitege 
vielen dank für den Tipp. Es löst zwar das Problem mathematisch, jedoch will mein Auftraggeber dass der Einzelpreis korrekt angezeigt wird. Beispiel 0,044€ - anstatt 44€. 
Wie wird dies bei Rechnungen mit Benzinpreisen gelöst ? Hier sind es ja auch mehr als 2 Nachkommastellen.

**Comment by tobitege** [comment]
In dem Bereich hatte ich noch keine Anforderungen (bis jetzt).
Vielleicht müsste man mal eine Testrechnung mit 4 Nachkommastellen an den Kunden schicken, ob der diese so aktzeptiert und auch verarbeiten kann?
Wäre m.E. nicht 100% gemäß dem Standard, aber in "echten Leben" muss der ja auch funktionieren. 😃

**Comment by TietjenMichael** [comment]
Vielen Dank,
ich muss somit das Feedback der Kunden abwarten ob es zu Beschwerden kommt.
